ratis-shell/src/main/java/org/apache/ratis/shell/cli/sh/command/AbstractRatisCommand.java

```diff
@@ -0,0 +1,179 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ratis.shell.cli.sh.command;
+
+import org.apache.ratis.shell.RetryUtil;
+import org.apache.ratis.shell.cli.Command;
+import org.apache.ratis.shell.cli.RaftUtils;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Options;
+import org.apache.ratis.client.RaftClient;
+import org.apache.ratis.proto.RaftProtos.FollowerInfoProto;
+import org.apache.ratis.proto.RaftProtos.RaftPeerProto;
+import org.apache.ratis.proto.RaftProtos.RaftPeerRole;
+import org.apache.ratis.proto.RaftProtos.RoleInfoProto;
+import org.apache.ratis.protocol.RaftClientReply;
+import org.apache.ratis.protocol.RaftGroup;
+import org.apache.ratis.protocol.RaftGroupId;
+import org.apache.ratis.protocol.RaftPeer;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * The base class for all the ratis shell {@link Command} classes.
+ */
+public abstract class AbstractRatisCommand implements Command {
+  public static final String PEER_OPTION_NAME = "peers";
+  public static final String GROUPID_OPTION_NAME = "groupid";
+  public static final RaftGroupId DEFAULT_RAFT_GROUP_ID
+      = RaftGroupId.valueOf(UUID.fromString("1-1-1-1-1"));
+  private final PrintStream printStream;
+  private RaftGroup raftGroup;
+
+  protected AbstractRatisCommand(Context context) {
+    printStream = context.getPrintStream();
+  }
+
+  @Override
+  public int run(CommandLine cl) throws IOException {
+    List<InetSocketAddress> addresses = new ArrayList<>();
+    String peersStr = "";
+    if (cl.hasOption(PEER_OPTION_NAME)) {
+      peersStr = cl.getOptionValue(PEER_OPTION_NAME);
+    }
+    String[] peersArray = peersStr.split(",");
+    for (int i = 0; i < peersArray.length; i++) {
+      String[] hostPortPair = peersArray[i].split(":");
+      InetSocketAddress addr =
+          new InetSocketAddress(hostPortPair[0], Integer.parseInt(hostPortPair[1]));
+      addresses.add(addr);
+    }
+
+    RaftGroupId raftGroupIdFromConfig = DEFAULT_RAFT_GROUP_ID;
+    if (cl.hasOption(GROUPID_OPTION_NAME)) {
+      raftGroupIdFromConfig = RaftGroupId.valueOf(
+          UUID.fromString(cl.getOptionValue(GROUPID_OPTION_NAME)));
+    }
+
+    List<RaftPeer> peers = addresses.stream()
+        .map(addr -> RaftPeer.newBuilder()
+            .setId(RaftUtils.getPeerId(addr))
+            .setAddress(addr)
+            .build()
+        ).collect(Collectors.toList());
+    raftGroup = RaftGroup.valueOf(raftGroupIdFromConfig, peers);
+    try (final RaftClient client = RaftUtils.createClient(raftGroup)) {
+      RaftGroupId remoteGroupId;
```

Comment:
When raftGroupIdFromConfig != null (i.e. user has specified a group id), remoteGroupId should be set to raftGroupIdFromConfig.

Reply:
Thanks for the suggestion, this make the logic clearly.